pkg/service/service.go

```diff
@@ -1120,59 +1029,79 @@ func UpdateServicesWithOwnerReferences(client *kclient.Client, k8sComponents []d
 		if found {
 			continue
 		}
-		u.SetOwnerReferences(append(u.GetOwnerReferences(), ownerReference))
+		d.SetOwnerReferences(append(d.GetOwnerReferences(), ownerReference))
 
-		err = client.UpdateDynamicResource(group, version, resource, crdName, u)
+		err = client.UpdateDynamicResource(gvr.Resource.Group, gvr.Resource.Version, gvr.Resource.Resource, u.GetName(), d)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func getCRDName(crd map[string]interface{}) (string, bool) {
-	metadata, ok := crd["metadata"].(map[string]interface{})
-	if !ok {
-		return "", false
+func isLinkResource(kind string) bool {
+	return kind == "ServiceBinding"
+}
+
+// GetGVKFromUnstructured takes an unstructured struct and returns group, version and kind information as strings
+// It assumes group to be "core" for cases where metadata.apiVersion can't be split using forward slash, e.g. 'v1'
+func GetGVKFromUnstructured(u unstructured.Unstructured) schema.GroupVersionKind {
+	apiVersion := strings.Split(u.GetAPIVersion(), "/")
+	var group, version, kind string
+	if len(apiVersion) != 2 {
+		group = ""
+		version = apiVersion[0]
+	} else {
+		group = apiVersion[0]
+		version = apiVersion[1]
 	}
-	name, ok := metadata["name"].(string)
-	if !ok {
-		return "", false
+	kind = u.GetKind()
+	return schema.GroupVersionKind{
+		Group:   group,
+		Version: version,
+		Kind:    kind,
 	}
-	return name, true
 }
 
-func isLinkResource(kind string) bool {
-	return kind == "ServiceBinding"
-}
+// GetGVRFromUnstructured returns GVR from unstructured data
+func GetGVRFromUnstructured(client *kclient.Client, u unstructured.Unstructured) (*meta.RESTMapping, error) {
+	gvk := GetGVKFromUnstructured(u)
 
-// createOperatorService creates the given operator on the cluster
-// it returns the CR,Kind and errors
-func createOperatorService(client *kclient.Client, d *DynamicCRD, labels map[string]string, ownerReferences []metav1.OwnerReference) (string, string, error) {
-	cr, csv, err := GetCSV(client, d.OriginalCRD)
+	cfg, err := client.KubeConfig.ClientConfig()
 	if err != nil {
-		return "", "", err
+		return &meta.RESTMapping{}, err
 	}
 
-	var group, version, kind, resource string
-	for _, crd := range csv.Spec.CustomResourceDefinitions.Owned {
-		if crd.Kind == cr {
-			group, version, kind, resource, err = getGVKRFromCR(crd)
-			if err != nil {
-				return cr, "", err
-			}
-			break
-		}
+	dc, err := discovery.NewDiscoveryClientForConfig(cfg)
+	if err != nil {
+		return &meta.RESTMapping{}, err
 	}
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(dc))
+
+	return mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+}
 
-	// add labels to the CRD before creation
-	d.AddComponentLabelsToCRD(labels)
+// getOperatorGVRList creates a slice of GVRs that are provided by Operators (CSV)
+func getOperatorGVRList(client *kclient.Client) ([]meta.RESTMapping, error) {
```

Comment:
similar to above function, this seems to be doing something else than what name suggests

Reply:
https://github.com/openshift/odo/pull/4964/commits/8270bb639dbd38589082f73b590955c489cdddd7

Reply:
Comment still says something else then function name

Reply:
Addressed.